packages/core/workflows-sdk/src/helper/workflow-export.ts

```diff
@@ -466,3 +474,56 @@ export const exportWorkflow = <TData = unknown, TResult = unknown>(
   MedusaWorkflow.registerWorkflow(workflowId, exportedWorkflow)
   return exportedWorkflow as MainExportedWorkflow<TData, TResult>
 }
+
+function attachOnFinishReleaseEvents(
+  events: DistributedTransactionEvents = {},
+  eventGroupId: string,
+  flow: LocalWorkflow
+) {
+  const onFinish = events.onFinish
+
+  const wrappedOnFinish = async (args: {
+    transaction: DistributedTransaction
+    result?: unknown
+    errors?: unknown[]
+  }) => {
+    await onFinish?.(args)
+
+    const eventBusService = (flow.container as MedusaContainer).resolve(
+      ModuleRegistrationName.EVENT_BUS,
+      { allowUnregistered: true }
+    )
+
+    if (!eventBusService || !eventGroupId) {
+      return
+    }
+
+    const logger =
+      (flow.container as MedusaContainer).resolve(
+        ContainerRegistrationKeys.LOGGER,
+        { allowUnregistered: true }
+      ) || console
+
+    const { transaction } = args
+    const failedStatus = [TransactionState.FAILED, TransactionState.REVERTED]
+
+    if (failedStatus.includes(transaction.getState())) {
+      await eventBusService.clearGroupedEvents(eventGroupId).catch(() => {
+        logger.warn(`Failed to clear events for eventGroupId - ${eventGroupId}`)
+      })
```

Comment:
skipped the return here 

```suggestion
      await eventBusService.clearGroupedEvents(eventGroupId).catch(() => {
        logger.warn(`Failed to clear events for eventGroupId - ${eventGroupId}`)
      })
      
      return
```

Reply:
oups I ve removed it by mistake, readded it but differently as well 💪